include/alpaka/event/EventGenericThreads.hpp

```diff
@@ -316,6 +317,33 @@ namespace alpaka
     {
         namespace traits
         {
+            namespace generic
+            {
+                template<typename TDev>
+                ALPAKA_FN_HOST auto currentThreadWaitForDevice(
+                    TDev const & dev
+                )
+                ->void
```

Comment:
IMO for void we should not use trailing return values. Since we depend on C++14 we can skip the type if we use auto but in a void method/function it is a little bit risky if someone is adding a return.

```C++
                template<typename TDev>
￼                ALPAKA_FN_HOST void currentThreadWaitForDevice(
￼                    TDev const & dev
￼                )
```
@BenjaminW3 What do you think?

Reply:
I agree, that the trailing void looks a bit silly, but it was like this in the original code.

Would be happy to change it, but this is not the only place with a trialing void, e.g. https://github.com/jkelling/alpaka/blob/queuegeneric/include/alpaka/event/EventGenericThreads.hpp#L72 , https://github.com/jkelling/alpaka/blob/queuegeneric/include/alpaka/event/EventGenericThreads.hpp#L267 , https://github.com/jkelling/alpaka/blob/queuegeneric/include/alpaka/event/EventGenericThreads.hpp#L307 , https://github.com/jkelling/alpaka/blob/queuegeneric/include/alpaka/event/EventGenericThreads.hpp#L326 ...

Should all be changed like this?

Reply:
There are also trailing -> bool . What about those?

Reply:
I would keep it. We use trailing return types for everything. I prefer consistency.

Reply:
... code cleanup might be better as a separate PR, IMHO.

Reply:
My page was not updated to show @BenjaminW3's comment when I wrote mine. Marking this as resolved for this PR.